src/Topologies/dss.jl

```diff
@@ -134,13 +134,13 @@ function create_dss_buffer(
                     @show supportedvectortypes
                 end
                 @assert fieldtype <: supportedvectortypes
-                if fieldtype <: Geometry.Covariant12Vector
```

Comment:
Don't we still need to support DSS for `Covariant12Vector`?

Reply:
Yes; I'll update this to retain the `Covariant12Vector` support

Reply:
Added tests in the `terrain_warp` test file + cleaned up the stranded `return nothing` statement.